core/remote/selectionmodelserver.cpp

```diff
@@ -29,16 +29,37 @@
 #include "selectionmodelserver.h"
 #include "server.h"
 
+#include <QTimer>
+
 using namespace GammaRay;
 
 SelectionModelServer::SelectionModelServer(const QString& objectName, QAbstractItemModel* model, QObject* parent):
-  NetworkSelectionModel(objectName, model, parent)
+  NetworkSelectionModel(objectName, model, parent), m_timer(new QTimer(this))
 {
+  m_timer->setSingleShot(true);
+  m_timer->setInterval(125);
+  Q_ASSERT(model);
+  // We do use a timer to group requests to avoid network overhead
+  connect(model, SIGNAL(rowsInserted(QModelIndex,int,int)), m_timer, SLOT(start()));
+  connect(model, SIGNAL(columnsInserted(QModelIndex,int,int)), m_timer, SLOT(start()));
+  connect(model, SIGNAL(rowsMoved(QModelIndex,int,int,QModelIndex,int)), m_timer, SLOT(start()));
+  connect(model, SIGNAL(rowsRemoved(QModelIndex,int,int)), m_timer, SLOT(start()));
+  connect(model, SIGNAL(columnsRemoved(QModelIndex,int,int)), m_timer, SLOT(start()));
+  connect(model, SIGNAL(columnsMoved(QModelIndex,int,int,QModelIndex,int)), m_timer, SLOT(start()));
```

Comment:
I guess i could remove the XXXRemoved one.
Others might be needed, in fact the slots slotCurrentChanged / slotCurrentColumnChanged / slotCurrentRowChanged does not propagate original passed flags, but hard fixed one.
This can lead to 'lost' flags, like the 'Clear' one etc, unfortunately not all members in the QISM are virtual in Qt4, so we can't override them all.
The best we an do here is to request to send again the correct selection.